src/physics/em/detail/BremFinalStateHelper.hh

```diff
@@ -0,0 +1,63 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file BremFinalStateHelper.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "physics/base/Units.hh"
+#include "physics/base/Secondary.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the angular distribution of photon from e+/e- Bremsstrahlung.
+ *
+ */
+class BremFinalStateHelper
+{
+  public:
+    //!@{
+    //! Type aliases
+    using Energy   = units::MevEnergy;
+    using Mass     = units::MevMass;
+    using Momentum = units::MevMomentum;
+    //!@}
+
+  public:
+    // Construct from data
+    inline CELER_FUNCTION BremFinalStateHelper(const Energy&     inc_energy,
+                                               const Real3&      inc_direction,
+                                               const Momentum&   inc_momentum,
+                                               const Mass&       inc_mass,
+                                               const ParticleId& gamma_id);
+
+    // Update the final state for the given RNG and the photon energy
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine&      rng,
+                                                 const Energy gamma_energy,
+                                                 Secondary*   secondaries);
```

Comment:
These last two arguments should be part of the constructor as well: it's "given the incident particle data and the exiting gamma energy, what is one realization of the exiting interaction" (which will all have the same energy change but different directions)

Reply:
To make this easier as a really simple helper function you could also just write this whole thing as a simple struct with public data members, then construct and call it like:
```c++
BremInteractionSampler sample_interaction{inc_energy, inc_direction, inc_momentum, inc_mass, gamma_energy, gamma_id};
return sample_interaction(rng);
```
Or you could get fancier and keep it as a class but create sub-distributions as class members, precalculating the exit energy and having the various samplers as member data. The only *real* advantage to doing it that way, though, would be to reduce the size of templated code if we had multiple RNG instantiations, which in practice we won't have. You could also argue that it improves the structure by separating the "setup" based on knowns from the "sampling."

Reply:
I will take your comment below.